src/runners/alert_handler.py

```diff
@@ -57,54 +57,87 @@ def log_failure(ctx, alert, e):
-            continue
-        print('Creating ticket for alert', alert)
-
-        # Create a new ticket in JIRA for the alert
-        try:
-            ticket_id = create_jira.create_jira_ticket(
-                alert_id=alert['ALERT_ID'],
-                query_id=alert['QUERY_ID'],
-                query_name=alert['QUERY_NAME'],
-                environment=alert['ENVIRONMENT'],
-                sources=alert['SOURCES'],
-                actor=alert['ACTOR'],
-                object=alert['OBJECT'],
-                action=alert['ACTION'],
-                title=alert['TITLE'],
-                event_time=alert['EVENT_TIME'],
-                alert_time=alert['ALERT_TIME'],
-                description=alert['DESCRIPTION'],
-                detector=alert['DETECTOR'],
-                event_data=alert['EVENT_DATA'],
-                severity=alert['SEVERITY'])
+            correlation_id = row[7]
         except Exception as e:
-            log.error(e, f"Failed to create ticket for alert {alert}")
-            log_failure(ctx, alert, e)
+            log.error("No CORRELATION_ID in alert", e)
+            try:
+                ticket_id = create_jira.create_jira_ticket(alert)
+            except Exception as e:
+                log.error(e, f"Failed to create ticket for alert {alert}")
+                log_failure(ctx, alert, e)
+                continue
             continue
+        log.info('Creating ticket for alert', alert)
+
+        # We check against the correlation ID for alerts in that correlation with the same ticket
+        query = f"""SELECT * from {ALERTS_TABLE} where CORRELATION_ID = '{correlation_id}' and TICKET is not null
+                    ORDER BY EVENT_TIME DESC
+                    LIMIT 1
+                """
+        correlated_results = ctx.cursor().execute(query).fetchall() if correlation_id else []
+
+        log.info(f"Discovered {len(correlated_results)} correlated results")
+
+        if len(correlated_results) > 0:
+
+            # There is a correlation with a ticket that exists, so we should append to that ticket
+            ticket_id = correlated_results[0][3]
+            ticket_status = create_jira.check_ticket_status(ticket_id)
+
+            if ticket_status == 'To Do':
+                try:
+                    create_jira.append_to_body(ticket_id, alert)
+                except Exception as e:
+                    log.error(f"Failed to append alert {alert['ALERT_ID']} to ticket {ticket_id}.", e)
+                    try:
+                        ticket_id = create_jira.create_jira_ticket(alert)
+                    except Exception as e:
+                        log.error(e, f"Failed to create ticket for alert {alert}")
+                        log_failure(ctx, alert, e)
+                    continue
+            else:
+                # The ticket is already in progress, we shouldn't change it
+                # Create a new ticket in JIRA for the alert
```

Comment:
that's going to be something that I definitely want to do, but i'd also like to keep out of the correlation changes so that I only have to think about one thing at once.